example/advection/advection_package.cpp

```diff
@@ -302,6 +317,28 @@ void PreFill(MeshBlockData<Real> *rc) {
           v(out + n, k, j, i) = 1.0 - v(in + n, k, j, i);
         });
   }
+
+  // Fill the metadata::None var with index gymnastics.
+  if (test_metadata_none) {
+    const int nx1 = pmb->cellbounds.ncellsi(IndexDomain::interior);
+    const int nx2 = pmb->cellbounds.ncellsj(IndexDomain::interior);
+    const int nx3 = pmb->cellbounds.ncellsk(IndexDomain::interior);
+
+    // packing in principle unnecessary/convoluted here and just done for demonstration
+    std::vector<std::string> vars({"metadata_none_var"});
+    PackIndexMap imap;
+    const auto &v = rc->PackVariables(vars, imap);
+
+    const int ivar_lo = imap.get("metadata_none_var").first;
+    const int ivar_hi = imap.get("metadata_none_var").second;
+    pmb->par_for(
+        PARTHENON_AUTO_LABEL, 0, 2, 0, nx3, 0, nx2, 0, nx1,
+        KOKKOS_LAMBDA(const int n, const int k, const int j, const int i) {
+          v(ivar_lo, n, k, j, i) = n + k * j * i;
+          v(ivar_lo + 1, n, k, j, i) = 1 + n + k * j * i;
+          v(ivar_hi, n, k, j, i) = 2 + n + k * j * i;
```

Comment:
i just looked at the updated loop.
While I understand the fix going from mesh to meshbock, I now don't get the `ivar_lo`/`ivar_hi` indexing.
What's supposed to happen here? I could see this working if `ivar_lo`/`ivar_hi replace the `n` index.

Reply:
Any comment on this? ping @AstroBarker 

I'm happy to merge this without adding test to read the data (which would be great), but I'd like to understand this loop before merging as from my current understanding it might seem like being an unintended bug.

Reply:
I've updated the loop, removing the `ivar_lo/hi` stuff, which I agree was not necessary in the loop, and added a section to the hdf5 test that checks the contents of this variable. Sorry for the delay!

Reply:
Great, thanks!